lib/hls/hls_parser.js

```diff
@@ -3407,20 +3407,34 @@ shaka.hls.HlsParser = class {
     // We accomplish this in our DASH-y model by setting a presentation
     // delay of configured value, or 3 segments duration if not configured.
     // This will be the "live edge" of the presentation.
-    let presentationDelay;
+    let presentationDelay = 0;
     if (this.config_.defaultPresentationDelay) {
       presentationDelay = this.config_.defaultPresentationDelay;
     } else if (this.lowLatencyPresentationDelay_) {
       presentationDelay = this.lowLatencyPresentationDelay_;
     } else if (this.presentationDelay_) {
       presentationDelay = this.presentationDelay_;
     } else {
-      const playlistSegments = playlist.segments.length;
+      const totalSegments = playlist.segments.length;
       let delaySegments = this.config_.hls.liveSegmentsDelay;
-      if (delaySegments > (playlistSegments - 2)) {
-        delaySegments = Math.max(1, playlistSegments - 2);
+      if (delaySegments > (totalSegments - 2)) {
+        delaySegments = Math.max(1, totalSegments - 2);
+      }
+      for (let i = totalSegments - delaySegments; i < totalSegments; i++) {
+        const extinfTag = shaka.hls.Utils.getFirstTagWithName(
+            playlist.segments[i].tags, 'EXTINF');
+        if (extinfTag) {
+          const extinfValues = extinfTag.value.split(',');
+          const duration = Number(extinfValues[0]);
+          if (duration % 1 === 0) {
```

Comment:
Is there really any reason to have this check? `Math.ceil()` can be called on integers just fine.

Reply:
Fixed!